platform/command/new_command.go

```diff
@@ -69,3 +92,65 @@ func MakeNewCommandEndpoint(svc Service) endpoint.Endpoint {
 		return newCommandResponse{Payload: payload}, nil
 	}
 }
+
+type RawCommand struct {
+	UDID        string `json:"udid" plist:"-"`
+	CommandUUID string `json:"command_uuid"`
+	Command     struct {
+		RequestType string `json:"request_type"`
+	} `json:"command"`
+	Raw []byte `plist:"-" json:"payload"`
+}
+
+type newRawCommandRequest struct {
+	RawCommand
+}
+
+type newRawCommandResponse struct {
+	Payload *RawCommand `json:"payload,omitempty"`
+	Err     error       `json:"error,omitempty"`
+}
+
+func (r newRawCommandResponse) Failed() error   { return r.Err }
+func (r newRawCommandResponse) StatusCode() int { return http.StatusCreated }
+
+func decodeNewRawCommandRequest(ctx context.Context, r *http.Request) (interface{}, error) {
+	udid, ok := mux.Vars(r)["udid"]
+	if !ok {
+		return nil, errors.New("empty udid")
+	}
+
+	payload, err := io.ReadAll(r.Body)
+	if err != nil {
+		return nil, fmt.Errorf("read payload body: %w", err)
+	}
+
+	// verify body is valid plist and parse CommandUUID and RequestType
+	var req newRawCommandRequest
+	if err := plist.NewXMLDecoder(bytes.NewBuffer(payload)).Decode(&req); err != nil {
```

Comment:
any reason not to just directly decode the body rather than buffer it? I guess to log it? it'll still generate an error if we parse it directly tho?

```suggestion
	if err := plist.NewXMLDecoder(r.Body).Decode(&req); err != nil {
```

Reply:
While I can't find anywhere that says this explictly, I believe MDM doesn't support binary plist commands? So this is explicitly only parsing XML plists (and also skipping the binary plist check that `Unmarshal` has).

Reply:
Ah, I see. As an extra-implicit check? Fair 'nuff.

Complete aside: Profiles (at least the initial OTA profile) can be bplists. A popular vendor uses that. fwiw.